### PR TITLE
add image service

### PR DESCRIPTION
**Pull request by AlessiaYChen**

### CODE REVIEW

File: services/net/image/ImageAction.cs [hunk 1]
```diff
@@ -68,58 +70,132 @@ public ImageAction(IKafkaMessenger kafka, IApiService api, IOptions<ImageOptions
...
+                        {
+                            // If another process has it in progress only attempt to do an import if it's
+                            // more than an 2 minutes old. Assumption is that it is stuck.
+                            reference = await this.Api.UpdateContentReferenceAsync(reference);
+                        }
+                        else sendMessage = false;
+
+                        if (reference != null)
+                        {
+                            tasks.Add(CopyImage(client, manager.DataSource, file.Name));
```
[the remark this answers is not in the document]
Comment (Fosol) [reply services/net/image/ImageAction.cs:121]: My recommendation is to not bother with creating multiple threads at this point.

File: services/net/image/ImageAction.cs [hunk 2]
```diff
@@ -212,5 +306,29 @@ protected DateTime GetLocalDateTime(DataSourceModel dataSource, DateTime date)
     {
         return date.ToTimeZone(DataSourceIngestManager<ImageOptions>.GetTimeZone(dataSource, this.Options.TimeZone));
     }
+
+    /// <summary>
+    /// map front page image name with datasource code
+    /// </summary>
+    /// <param name="dataSource"></param>
+    /// <returns></returns>
+    private string MappingImageFile(DataSourceModel dataSource)
```
Comment (Fosol) [inline services/net/image/ImageAction.cs:315]: This isn't ideal.  If we need to match a data source with an image naming convention.  Either the data source `Code` should be appropriate, or the `DataSource.Connection` should contain a relevant configuration value.

File: services/net/image/appsettings.json [hunk 3]
```diff
@@ -15,7 +15,11 @@
     "OutputPath": "/data/image",
     "Command": "ffmpeg",
     "MediaTypes": "Image",
-    "PrivateKeysPath": "keys"
+    "PrivateKeysPath": "keys",
+    "PrivateKeyFileName": "id_rsa",
+    "InputPath": "/dsk98/binaryroot/",
+    "HostName": "scharnhorst.tno.gov.bc.ca",
```
Comment (Fosol) [inline services/net/image/appsettings.json:21]: I wouldn't include the `HostName` or `UserName` in the default config values.  They're only for local testing.

**Review by Fosol — CHANGES_REQUESTED** [review]

File: services/net/image/ImageAction.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (AlessiaYChen) [reply services/net/image/ImageAction.cs]: add await for each CopyImage task

**Review by Fosol — APPROVED** [review]

**Pull request merged**